app/models/topic.rb

```diff
@@ -655,16 +654,17 @@ def add_moderator_post(user, text, opts=nil)
                               skip_validations: true,
                               custom_fields: opts[:custom_fields])
 
-    if (new_post = creator.create) && new_post.present?
-      increment!(:moderator_posts_count) if new_post.persisted?
-      # If we are moving posts, we want to insert the moderator post where the previous posts were
-      # in the stream, not at the end.
-      new_post.update_attributes!(post_number: opts[:post_number], sort_order: opts[:post_number]) if opts[:post_number].present?
+    new_post = creator.create
 
-      # Grab any links that are present
-      TopicLink.extract_from(new_post)
-      QuotedPost.extract_from(new_post)
-    end
+    increment!(:moderator_posts_count) if new_post.persisted?
+
+    # If we are moving posts, we want to insert the moderator post where the previous posts were
+    # in the stream, not at the end.
+    new_post.update_columns(post_number: opts[:post_number], sort_order: opts[:post_number]) if opts[:post_number].present?
```

Comment:
Ah, I forgot. I also added one extra simplification in https://github.com/discourse/discourse/pull/4691/commits/6e14eb33f7a6ea59abbedfcd9eeb149d248c4932. The idea is that by checking `ignore_validations?` at a single place we won't have any more bugs similar to the one fixed by https://github.com/discourse/discourse/pull/4691/commits/aa5b98ed09a384074deb0b192e889d2f48aa5782.